### PR TITLE
resolves #4, resolves #9, resolves #10, resolves #11 

### LINKED ISSUE #4: Scripts Have Windows Line Endings
The scripts don't start on linux because they use Windows file endings:

    ./cordaptor.sh
    bash: ./cordaptor.sh: /bin/sh^M: bad interpreter: No such file or directory
    
    file cordaptor.sh
    cordaptor.sh: POSIX shell script, ASCII text executable, with CRLF line terminators

### LINKED ISSUE #9: java.time.Duration serialisation is not supported
Steps to reproduce:
1. Run cordaptor v 0.1.0 with obligation cordapp (Advanced examples from corda samples repo) in embedded mode
2. Open the api.json page

While trying to serialize an object with java.time.Duration class as a property, there is a serialisation exception thrown.

`{
  "cause": {
    "class": "tech.b180.cordaptor.rest.SerializationException",
    "message": "Cannot create a serializer for type Erased(net.corda.finance.contracts.asset.Obligation$State) introspected as non-composable for the following reason: Has properties [template] of types that are not serializable:\ntemplate [net.corda.finance.contracts.asset.Obligation$Terms<*>]: Has properties [timeTolerance] of types that are not serializable:\n    timeTolerance [class java.time.Duration]: Mandatory constructor parameters [arg0, arg1] are missing from the readable properties []",
    "cause": "null"
  },
  "errorType": "GENERIC_ERROR",
  "message": "Unknown server error"
}`

### LINKED ISSUE #10: Serialisation of NonEmptySet causes an error
Steps to reproduce:

1. Run cordaptor v 0.1.0 with obligation cordapp (Advanced examples from corda samples repo) in embedded mode
2. Open the api.json page

Serialisation of corda core type NonEmptySet is not supported and throws an error - 
```
{
  "cause": {
    "class": "tech.b180.cordaptor.rest.SerializationException",
    "message": "Error finding serializer for property acceptableContracts of object Parameterised(Terms<*>)",
    "cause": "java.lang.AssertionError: Don't know how to make instances of class net.corda.core.utilities.NonEmptySet"
  },
  "errorType": "GENERIC_ERROR",
  "message": "Unknown server error"
}
```

NonEmptySet is introspected as a Collection type and the solution would handle all such non supported collection types that do not need to be de-serialised.

### LINKED ISSUE #11: Flow classes not extending FlowLogic directly are not serialised
Steps to reproduce:

1. Run cordaptor v 0.1.0 with obligation cordapp (Advanced examples from corda samples repo) in standalone mode
2. Open the api.json page
3. CashFlowExit flow is not serialised and causes an exception

```
cordaptor_1          | Exception in thread "main" java.lang.AssertionError: Flow class class net.corda.finance.flows.CashExitFlow does not seem to extend FlowLogic
cordaptor_1          |  at tech.b180.cordaptor.corda.ReflectionUtilsKt.determineFlowResultClass(ReflectionUtils.kt:16)
```

The fix will involve introspecting all supertypes rather than direct supertype when introspecting flow classes

### PR DESCRIPTION
**Pull request by parthbond180**
fixes for various issues based on testing with obligation cordapp

Change summary:
1. changed min platform version to 6 from 7 as min supported version of corda core tested against is corda 4.4 (platform version 6)
2. Added serialiser for Duration
3. Modified reflection logic to enable serailisation for types extending FlowLogic  
4. Added line endings for UNIX

### CODE REVIEW

File: bundle-rest-embedded/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ apply plugin: 'net.corda.plugins.cordapp'
 
 cordapp {
     targetPlatformVersion 7
-    minimumPlatformVersion 7
+    minimumPlatformVersion 6
```
Comment (igor-b180) [inline bundle-rest-embedded/build.gradle:6]: Worth adding a comment explaining why it's the minimum version

File: rest-endpoint/src/main/kotlin/tech/b180/cordaptor/rest/CordaTypesSerializers.kt [hunk 2]
```diff
@@ -208,6 +209,27 @@ class JavaInstantSerializer : CustomSerializer<Instant>,
   }
 }
 
+/**
+ * Serializer for an [Duration] representing it as a JSON string value formatted as an ISO-8601 timestamp.
```
Comment (igor-b180) [inline rest-endpoint/src/main/kotlin/tech/b180/cordaptor/rest/CordaTypesSerializers.kt:213]: Misleading comment about 'timestamp', it's actually ISO duration format

File: rest-endpoint/src/main/kotlin/tech/b180/cordaptor/rest/CordaTypesSerializers.kt [hunk 3]
```diff
@@ -208,6 +209,27 @@ class JavaInstantSerializer : CustomSerializer<Instant>,
...
+        SerializationFactory.DelegatingSerializer<Duration, String>(
+                delegate = SerializationFactory.StringSerializer,
+                my2delegate = { this.toString() },
+                delegate2my = { Duration.parse(it) }
+        ) {
+
+  override fun generateSchema(generator: JsonSchemaGenerator): JsonObject {
+    return mapOf(
+            "type" to "string",
+            "format" to "date-time"
```
Comment (igor-b180) [inline rest-endpoint/src/main/kotlin/tech/b180/cordaptor/rest/CordaTypesSerializers.kt:227]: The format is not date-time, but ISO duration

File: bundle-rest-embedded/build.gradle (hunk 1 above)
Comment (parthbond180) [reply bundle-rest-embedded/build.gradle:6]: Updated commit message

**Pull request merged**